---
 Straight forward fix to allow shutdown on systems not using policykit.

 If the system really don't have policykit, it will make things work
 as in 2.22.
 
 If pk is in gsm but not in ck, it'll still work as in 2.22.

 If pk is present in both, you just have to fix the logic to make
 gsm ask pk if the user has the permission to shutdown _before_ showing
 buttons and in case this is not possible, just damn fix pk.

 reported as http://bugzilla.gnome.org/show_bug.cgi?id=549150

 gnome-session/gsm-consolekit.c |   22 ++++++++++------------
 1 file changed, 10 insertions(+), 12 deletions(-)

Index: b/gnome-session/gsm-consolekit.c
===================================================================
--- a/gnome-session/gsm-consolekit.c
+++ b/gnome-session/gsm-consolekit.c
@@ -528,7 +528,7 @@ request_restart_priv (GsmConsolekit *man
                 g_free (error_message);
         }
 #else
-        g_assert_not_reached ();
+        g_warning ("Failed to restart and Policykit support not enabled");
 #endif /* HAVE POLKIT */
 }
 
@@ -568,7 +568,7 @@ request_stop_priv (GsmConsolekit *manage
                 }
         }
 #else
-        g_assert_not_reached ();
+        g_warning ("Failed to shutdown and Policykit support not enabled");
 #endif /* HAVE POLKIT */
 }
 
@@ -877,12 +877,19 @@ gsm_consolekit_can_do_action (GsmConsole
 
         return result != POLKIT_RESULT_NO && result != POLKIT_RESULT_UNKNOWN;
 }
+#else
+static gboolean
+gsm_consolekit_can_do_action (GsmConsolekit *manager,
+                              const char    *action_id)
+{
+        /* if we got here, res was true */
+        return TRUE;
+}
 #endif
 
 gboolean
 gsm_consolekit_can_restart (GsmConsolekit *manager)
 {
-#ifdef HAVE_POLKIT_GNOME
         gboolean res;
         GError  *error;
 
@@ -897,16 +904,11 @@ gsm_consolekit_can_restart (GsmConsoleki
 
         return gsm_consolekit_can_do_action (manager, "org.freedesktop.consolekit.system.restart") ||
                gsm_consolekit_can_do_action (manager, "org.freedesktop.consolekit.system.restart-multiple-users");
-#else
-        g_debug ("GsmConsolekit: built without PolicyKit-gnome support - cannot restart system");
-        return FALSE;
-#endif
 }
 
 gboolean
 gsm_consolekit_can_stop (GsmConsolekit *manager)
 {
-#ifdef HAVE_POLKIT_GNOME
         gboolean res;
         GError  *error;
         error = NULL;
@@ -920,10 +922,6 @@ gsm_consolekit_can_stop (GsmConsolekit *
 
         return gsm_consolekit_can_do_action (manager, "org.freedesktop.consolekit.system.stop") ||
                gsm_consolekit_can_do_action (manager, "org.freedesktop.consolekit.system.stop-multiple-users");
-#else
-        g_debug ("GsmConsolekit: built without PolicyKit-gnome support - cannot stop system");
-        return FALSE;
-#endif
 }
 
 GsmConsolekit *
